cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/purgatory/RequestInfo.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2019 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License"). See License in the project root for license information.
+ */
+
+package com.linkedin.kafka.cruisecontrol.servlet.purgatory;
+
+import com.linkedin.kafka.cruisecontrol.servlet.EndPoint;
+import com.linkedin.kafka.cruisecontrol.servlet.parameters.CruiseControlParameters;
+import com.linkedin.kafka.cruisecontrol.servlet.parameters.ParameterUtils;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import javax.servlet.http.HttpServletRequest;
+
+import static com.linkedin.kafka.cruisecontrol.servlet.KafkaCruiseControlServletUtils.getClientIpAddress;
+import static com.linkedin.kafka.cruisecontrol.servlet.purgatory.ReviewStatus.*;
+
+
+/**
+ * A class to represent request information in purgatory. Possible status of requests, with supported transitions:
+ * <ul>
+ * <li>{@link ReviewStatus#PENDING_REVIEW} -> {@link ReviewStatus#APPROVED}, {@link ReviewStatus#DISCARDED}</li>
+ * <li>{@link ReviewStatus#APPROVED} -> {@link ReviewStatus#DISCARDED}, {@link ReviewStatus#SUBMITTED}</li>
+ * </ul>
+ */
+public class RequestInfo {
+  private static final String INIT_REASON = "Awaiting review.";
+  private static final Map<ReviewStatus, Set<ReviewStatus>> VALID_TRANSFER = new HashMap<>();
+  static {
+    VALID_TRANSFER.put(PENDING_REVIEW, new HashSet<>(Arrays.asList(APPROVED, DISCARDED)));
+    VALID_TRANSFER.put(APPROVED, new HashSet<>(Arrays.asList(DISCARDED, SUBMITTED)));
```

Comment:
do we allow requester to discard a request?

Reply:
Yes, this is intentional -- i.e. an approved request can be discarded later.